### ISSUE TITLE
020720 Add possibility to Center the text In the TextField Component or add other advantage methods to TextArea

### ISSUE THREAD

**Issue by DurankGts** [codenameone/CodenameOne#3149]
Please create a way to Center the Text in the TextField Component because this component have other method very importants that don't have TextArea Component that permitt Center the Text in CENTER.

This are the advantages: 
TextField Component
*Center Text after edit = Don't supported
*addDataChangedListener method = supported
*DoneListener = supported

TextArea Component
*Center Text after edit = supported
*addDataChangedListener method = Don't supported
*DoneListener = don't supported

**Comment by jsfan3** [comment]
It sounds to my mind that what you are asking for is already available with this simple workaround, which creates a TextField using a TextArea. Try this code, it works for me in the simulator:

```
        Form hi = new Form("Hi World", BoxLayout.y());
        TextArea textField = new TextArea();
        textField.setRows(1);
        textField.setGrowByContent(false);
        textField.setGrowLimit(1);
        hi.add(textField);
        textField.getAllStyles().setAlignment(Component.CENTER);
        textField.setDoneListener(l -> {
            Log.p("Done listener invoked");
        });
        textField.addDataChangedListener((int type, int index) -> {
            Log.p("Data changed listener invoked");
        });
        hi.show();
```

**Comment by codenameone** [comment]
Actually CENTER is only supported for display not editing so this will "jump" to the left or right when editing.
This is a pretty difficult thing to implement as native editing aligned to the center will cause a noticeable "jump" between our position and the native position.

**Comment by DurankGts** [comment]
addDataChangedListener is not supported by TextArea component.

**Comment by jsfan3** [comment]
Why do you say that “addDataChangedListener is not supported by TextArea component“? Have you tried the code I posted? It works for me, see the logs while you are typing in the TextArea.

**Comment by DurankGts** [comment]
yes, in the last update of codenameone source code is not' supported
![image](https://user-images.githubusercontent.com/16245755/87048268-85775900-c1c9-11ea-93bb-aa28ec87d82d.png)

**Comment by jsfan3** [comment]
Maybe you are not using the last version of Codename One. As you can see in the class https://github.com/codenameone/CodenameOne/blob/master/CodenameOne/src/com/codename1/ui/TextArea.java there is the method:
```
**
     * Adds a listener for data change events it will be invoked for every change
     * made to the text field, notice most platforms will invoke only the 
     * DataChangedListener.CHANGED event
     * 
     * @param d the listener
     */
    public void addDataChangedListener(DataChangedListener d) {
        listeners.addListener(d);
    }
```

**Comment by DurankGts** [comment]
thank you
